### PR TITLE
kong: bump Kong to 1.4.0

### PR DESCRIPTION
**Pull request by kikito**

### CODE REVIEW

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index ae384ce..dd8312a 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -4,10 +4,10 @@ kong:1.1 @ amd64
 kong:1.1-centos @ amd64
 kong:1.2 @ amd64
 kong:1.2-centos @ amd64
-kong:1.4rc2 @ amd64
-kong:1.4rc2-centos @ amd64
-kong:1.4rc2-ubuntu @ amd64
-kong:1.4rc2-ubuntu @ arm64v8
+kong:1.4 @ amd64
+kong:1.4-centos @ amd64
+kong:1.4-ubuntu @ amd64
+kong:1.4-ubuntu @ arm64v8
 kong:centos @ amd64
 kong:latest @ amd64
 kong:ubuntu @ amd64
diff --git a/_bashbrew-list b/_bashbrew-list
index b24c9f2..f5a88dd 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -20,13 +20,13 @@ kong:1.3.0
 kong:1.3.0-alpine
 kong:1.3.0-centos
 kong:1.3.0-ubuntu
-kong:1.4rc2
-kong:1.4rc2-centos
-kong:1.4rc2-ubuntu
-kong:1.4.0rc2
-kong:1.4.0rc2-alpine
-kong:1.4.0rc2-centos
-kong:1.4.0rc2-ubuntu
+kong:1.4
+kong:1.4-centos
+kong:1.4-ubuntu
+kong:1.4.0
+kong:1.4.0-alpine
+kong:1.4.0-centos
+kong:1.4.0-ubuntu
 kong:alpine
 kong:centos
 kong:latest
diff --git a/kong_1.4rc2-centos/Dockerfile b/kong_1.4-centos/Dockerfile
similarity index 97%
rename from kong_1.4rc2-centos/Dockerfile
rename to kong_1.4-centos/Dockerfile
index 496bdc5..ccdd788 100644
--- a/kong_1.4rc2-centos/Dockerfile
+++ b/kong_1.4-centos/Dockerfile
@@ -1,7 +1,7 @@
 FROM centos:7
 LABEL maintainer="Kong Core Team <team-core@konghq.com>"
 
-ENV KONG_VERSION 1.4.0rc2
+ENV KONG_VERSION 1.4.0
 
 ARG SU_EXEC_VERSION=0.2
 ARG SU_EXEC_URL="https://github.com/ncopa/su-exec/archive/v${SU_EXEC_VERSION}.tar.gz"
diff --git a/kong_1.4rc2-centos/docker-entrypoint.sh b/kong_1.4-centos/docker-entrypoint.sh
similarity index 100%
rename from kong_1.4rc2-centos/docker-entrypoint.sh
rename to kong_1.4-centos/docker-entrypoint.sh
diff --git a/kong_1.4rc2-ubuntu/Dockerfile b/kong_1.4-ubuntu/Dockerfile
similarity index 96%
rename from kong_1.4rc2-ubuntu/Dockerfile
rename to kong_1.4-ubuntu/Dockerfile
index b409900..82320fd 100644
--- a/kong_1.4rc2-ubuntu/Dockerfile
+++ b/kong_1.4-ubuntu/Dockerfile
@@ -1,7 +1,7 @@
 FROM ubuntu:xenial
 LABEL maintainer="Kong Core Team <team-core@konghq.com>"
 
-ENV KONG_VERSION 1.4.0rc2
+ENV KONG_VERSION 1.4.0
 
 RUN apt-get update \
     && apt-get install -y --no-install-recommends ca-certificates curl perl unzip \
diff --git a/kong_1.4rc2-ubuntu/docker-entrypoint.sh b/kong_1.4-ubuntu/docker-entrypoint.sh
similarity index 100%
rename from kong_1.4rc2-ubuntu/docker-entrypoint.sh
rename to kong_1.4-ubuntu/docker-entrypoint.sh
diff --git a/kong_1.4rc2/Dockerfile b/kong_1.4/Dockerfile
similarity index 89%
rename from kong_1.4rc2/Dockerfile
rename to kong_1.4/Dockerfile
index eaf6fdd..469ccb8 100644
--- a/kong_1.4rc2/Dockerfile
+++ b/kong_1.4/Dockerfile
@@ -1,8 +1,9 @@
 FROM alpine:3.10
 LABEL maintainer="Kong Core Team <team-core@konghq.com>"
 
-ENV KONG_VERSION 1.4.0rc2
-ENV KONG_SHA256 36da7808037be8e9299096e0042d3a720e729492a5b8f9fcfde64c909893b0be
+ENV KONG_VERSION 1.4.0
+ENV KONG_SHA256 e31080487cfb8262a839c19d5e8fdb18b7686e9e4cc594803eaa565e0b34ff7a
+
 
 RUN adduser -Su 1337 kong \
 	&& mkdir -p "/usr/local/kong" \
diff --git a/kong_1.4rc2/docker-entrypoint.sh b/kong_1.4/docker-entrypoint.sh
similarity index 100%
rename from kong_1.4rc2/docker-entrypoint.sh
rename to kong_1.4/docker-entrypoint.sh
```

</details>

**Comment by yosifkit** [comment]
Build test of #6846; d9864068; `amd64` (`kong`):

```console

$ bashbrew build kong:1.4.0-alpine
Building bashbrew/cache:debb2387da7bc9813627d871f2f2d92fa90ca7d1a70752fbfc45da5076c69119 (kong:1.4.0-alpine)
Tagging kong:1.4.0-alpine
Tagging kong:1.4.0
Tagging kong:1.4

$ test/run.sh kong:1.4.0-alpine
testing kong:1.4.0-alpine
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.4.0-ubuntu
Building bashbrew/cache:241bcf0d6096d3a4673a45073902315bf4f9ed11b82318fb11539cff370056d5 (kong:1.4.0-ubuntu)
Tagging kong:1.4.0-ubuntu
Tagging kong:1.4-ubuntu

$ test/run.sh kong:1.4.0-ubuntu
testing kong:1.4.0-ubuntu
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.4.0-centos
Building bashbrew/cache:6458d2974c42d42fc154cbebbac456674c25011bd16a8b85200f0c7a0880f086 (kong:1.4.0-centos)
Tagging kong:1.4.0-centos
Tagging kong:1.4-centos

$ test/run.sh kong:1.4.0-centos
testing kong:1.4.0-centos
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'root': 'locked'
passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.3.0-alpine
Building bashbrew/cache:ef7f2cd00816dcc505c6296007efa6ba33faca1aa44dce0f36902ec6c7875b44 (kong:1.3.0-alpine)
Tagging kong:1.3.0-alpine
Tagging kong:1.3.0
Tagging kong:1.3
Tagging kong:alpine
Tagging kong:latest

$ test/run.sh kong:1.3.0-alpine
testing kong:1.3.0-alpine
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.3.0-ubuntu
Building bashbrew/cache:027cd79ed6f8f964e61e257c12a5fe7e06a75ac98ab5e585bac42fe528bb46f8 (kong:1.3.0-ubuntu)
Tagging kong:1.3.0-ubuntu
Tagging kong:1.3-ubuntu
Tagging kong:ubuntu

$ test/run.sh kong:1.3.0-ubuntu
testing kong:1.3.0-ubuntu
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.3.0-centos
Using bashbrew/cache:18b826a87f2e1aa129acdb9c8aeda2065b8651eb04e94a746bc824da156e0468 (kong:1.3.0-centos)
Tagging kong:1.3.0-centos
Tagging kong:1.3-centos
Tagging kong:centos

$ test/run.sh kong:1.3.0-centos
testing kong:1.3.0-centos
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'root': 'locked'
passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.2.2-alpine
Building bashbrew/cache:094407d6061f7f7cba6ceb24e38adf1cfe9da1173e7308bea0b9961802d7d38c (kong:1.2.2-alpine)
Tagging kong:1.2.2-alpine
Tagging kong:1.2.2
Tagging kong:1.2

$ test/run.sh kong:1.2.2-alpine
testing kong:1.2.2-alpine
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.2.2-centos
Using bashbrew/cache:9dfef0e73c7141cf99dd3e74853f4d8481b8977d0871c110010f538953c2dd5a (kong:1.2.2-centos)
Tagging kong:1.2.2-centos
Tagging kong:1.2-centos

$ test/run.sh kong:1.2.2-centos
testing kong:1.2.2-centos
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'root': 'locked'
passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.1.3-alpine
Building bashbrew/cache:1aff5ba95b01573f5a4c9098d7453c3003abb2cc9b6bc92abc81adb7d87486cc (kong:1.1.3-alpine)
Tagging kong:1.1.3-alpine
Tagging kong:1.1.3
Tagging kong:1.1

$ test/run.sh kong:1.1.3-alpine
testing kong:1.1.3-alpine
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.1.3-centos
Using bashbrew/cache:2499b59362731995b48de585d9640c2f753fe3ed2259415e65dd672f6758c894 (kong:1.1.3-centos)
Tagging kong:1.1.3-centos
Tagging kong:1.1-centos

$ test/run.sh kong:1.1.3-centos
testing kong:1.1.3-centos
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'root': 'locked'
passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.0.4-alpine
Building bashbrew/cache:5cbe5ebbaa2ba59990c2be9dd6c8d0c2622e936fb709fc557c6bb25fb07563da (kong:1.0.4-alpine)
Tagging kong:1.0.4-alpine
Tagging kong:1.0.4
Tagging kong:1.0

$ test/run.sh kong:1.0.4-alpine
testing kong:1.0.4-alpine
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build kong:1.0.4-centos
Using bashbrew/cache:065edf1a45e35086b1cf1f52c1965ba261218f1b95fc959b8b5f87d59de6db59 (kong:1.0.4-centos)
Tagging kong:1.0.4-centos
Tagging kong:1.0-centos

$ test/run.sh kong:1.0.4-centos
testing kong:1.0.4-centos
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...warning: garbage password detected for 'root': 'locked'
passed
	'override-cmd' [4/4]...passed

```

File: library/kong [hunk 1]
```diff
@@ -2,21 +2,21 @@ Maintainers: Kong Docker Maintainers <support@konghq.com> (@thekonginc)
 GitRepo: https://github.com/Kong/docker-kong.git
 # needs "Constraints" to match "centos" (which this image is "FROM")
 
-Tags: 1.4.0rc2-alpine, 1.4.0rc2, 1.4rc2
-GitCommit: 51ca0adb412814c84accc80b4fbc76b88b36dfe5
-GitFetch: refs/tags/1.4.0rc2
+Tags: 1.4.0-alpine, 1.4.0, 1.4
```
Comment (yosifkit) [inline library/kong]: After I merge, I realized that you may want to move the plain `alpine`, `latest`, `ubuntu`, and `centos` tags up from the `1.3` images?

Comment (kikito) [reply library/kong]: You are absolutely right, thanks for pointing this out.

I've made such change in #6855 

Apologies if this produces any inconvenience.

**Pull request merged**